stuff/cron/build_problem_rec_model.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/python3
+
+'''Builds the user recommendation model.
+
+This script reads user submissions and builds a recommendation model for
+which problems to attempt after a user has just sucessfully solved a problem.
+
+The model is a probability distribution of which problems a random user would
+attempt to solve next, based solely on the sequence of recently solved problems
+up to the latest one.
+'''
+
+import argparse
+import logging
+import sqlite3
+import warnings
+
+import numpy as np
+import pandas as pd
+
+import lib.db
+import lib.logs
+
+
+# Default training parameters.
+_TRAIN_FRACTION = 0.8
+_NUM_FOLLOWUPS = 3
+_FOLLOWUP_DECAY = 0.4
+
+
+def mean_average_precision(predicted, expected, k):
+    '''Compute MAP@k - https://goo.gl/KMghXR'''
+    if predicted is None or expected is None:
```

Comment:
nit: es un poquito más pythonic usar

```suggestion
    if not predicted or not expected:
```

Reply:
Done